packages/app/src/vis-packs/core/matrix/utils.ts

```diff
@@ -23,7 +23,7 @@ import { format } from 'd3-format';
 
 export function createNumericFormatter(
   notation: Notation,
-): ValueFormatter<NumericType> {
+): (val: ScalarValue<NumericType>) => string {
```

Comment:
I think using `ScalarValue` directly for the return types of formatter factories is more explicit.